### ISSUE TITLE
improvements request: --syntax-check should not prompt on vars_prompt

### ISSUE THREAD

**Issue by resmo** [ansible/ansible#13319]
##### Issue Type:

Bug Report / Improvement
##### Ansible Version:
- v2.0.0-0.6.rc1
- 1.9.4
##### Summary:

When running ansible-playbook with argument `--syntax-check` prompts are executed instead of only syntax checked
##### Steps To Reproduce:

``` yaml
- hosts: '{{ hosts }}'
  vars_prompt:
    - name: "hosts"
      prompt: "On which hosts you want to run it?"
      default: "none"
      private: no
  tasks:
    - debug: msg="test"
```
##### Expected Results:

Syntax check does not stop on prompts.
##### Actual Results:

```
ansible-playbook playbook.yml --syntax-check
On which hosts you want to run it? [none]: 
```

**Comment by hloeffler** [comment]
:+1:

**Comment by bcoca** [comment]
fixed for 2.0, it looks like it would require a much more intrusive change in 1.9, so i'm deferring that for now.

**Comment by resmo** [comment]
that is ok for me, expected only to be fixed in 2.x, but wanted to let you know, it is not a regression.
